python/cm/migrations/0092_auto_20220906_0957.py

```diff
@@ -0,0 +1,183 @@
+                parent=None,
+            )
+            service = ClusterObject.objects.get(prototype=service_prototype, cluster=cluster)
+            context["service"] = {"id": service.pk, "name": service.prototype.display_name}
+        elif task.action.prototype.type == "component":
+            service_prototype = Prototype.objects.get(
+                type=task.action.prototype.parent.type,
+                name=task.action.prototype.parent.name,
+                bundle=cluster.prototype.bundle,
+                version=cluster.prototype.version,
+                parent=None,
+            )
+            service = ClusterObject.objects.get(prototype=service_prototype, cluster=cluster)
+            component_prototype = Prototype.objects.get(
+                type=task.action.prototype.type,
+                name=task.action.prototype.name,
+                bundle=cluster.prototype.bundle,
+                version=cluster.prototype.version,
+                parent=service_prototype,
+            )
+            component = ServiceComponent.objects.get(
+                cluster=cluster,
+                service=service,
+                prototype=component_prototype,
+            )
+            context["service"] = {"id": service.pk, "name": service.prototype.display_name}
+            context["component"] = {
+                "id": component.pk,
+                "name": component.prototype.display_name,
+            }
+    except ObjectDoesNotExist:
+        context = {}
+    return context
+
+
+def _fix_selector(selector, _models):
+    fixed_selector = {}
+    for object_type, object_id in selector.items():
+        try:
+            model, name_attr = _models.get(object_type, (None, None))
+            if model:
+                obj = model.objects.get(pk=object_id)
+                if object_type in {"service", "component"}:
+                    obj = obj.prototype
+                fixed_selector[object_type] = {"id": object_id, "name": getattr(obj, name_attr)}
+        except ObjectDoesNotExist:
+            pass
+    return fixed_selector
+
+
+def get_selector(apps, schema_editor):
+    TaskLog = apps.get_model("cm", "TaskLog")
+
+    _models = {
+        "adcm": (apps.get_model("cm", "ADCM"), "name"),
+        "cluster": (apps.get_model("cm", "Cluster"), "name"),
+        "service": (apps.get_model("cm", "ClusterObject"), "display_name"),
+        "component": (apps.get_model("cm", "ServiceComponent"), "display_name"),
+        "provider": (apps.get_model("cm", "HostProvider"), "name"),
+        "host": (apps.get_model("cm", "Host"), "fqdn"),
+    }
+
+    for task in TaskLog.objects.all():
+
+        if task.selector:
+            selector = _fix_selector(task.selector, _models)
+            task.selector = selector
+            task.joblog_set.filter().update(selector=selector)
+
+        else:
+            try:
+                model, _ = _models.get(MODEL_TYPE[task.object_type.model], (None, None))
```

Comment:
Can be out of `try` checked by `if ... not in ...: continue`

Reply:
Fixed